mtp_api/apps/credit/managers.py

```diff
@@ -65,6 +65,11 @@ def update_prisons(self):
                 (CREDIT_RESOLUTION.PENDING,)
             )
 
+    def create(self, *args, **kwargs):
+        credit_instance = super(CreditManager, self).create(*args, **kwargs)
+        credit_instance.attach_profiles()
```

Comment:
tracing this method back to where it's called from for…
[transactions](https://github.com/ministryofjustice/money-to-prisoners-api/blob/0551de9982f0a9c13316603c18af2afc1b64c1e8/mtp_api/apps/transaction/serializers.py#L26) – this is fine, all the information a credit may have will already exist.
[payments](https://github.com/ministryofjustice/money-to-prisoners-api/blob/0551de9982f0a9c13316603c18af2afc1b64c1e8/mtp_api/apps/payment/serializers.py#L76) – i'm pretty sure this is too early. sender information is not known until the payment is captured so the sender profile will always be fixed as anonymous, won't it?

Reply:
Ultimately it's going to be calling through to https://github.com/ministryofjustice/money-to-prisoners-api/blob/0551de9982f0a9c13316603c18af2afc1b64c1e8/mtp_api/apps/security/managers.py#L114 so it depends if the billing address on the payment will be populated at this point. If it's not caught here, then the `attach_profiles_for_legacy_credits` method on `update_security_profiles` will create the sender profile